### DISCUSSION TITLE
Migration 1.30.11 -> 2.25

### DISCUSSION THREAD

**Discussion by mr2017x** [kimai/kimai#5204] [On-Premise: Installation & Updates] [answered]
### Describe the issue

Hi,

thanks for the software, i really like and want to jump now from 1.30 to 2.x

But when moving with migration skript it says:

  An exception occurred while executing a query: SQLSTATE[42S22]: Column not found: 1054 Unknown column 'option' in 'where clause'

how can i fix this?

### I already tried

- [X] I've read and searched [the documentation](https://www.kimai.org/documentation/).
- [X] I've searched for similar issues in this repository.
- [X] I've searched for similar issues in [the discussions](https://github.com/kimai/kimai/discussions).

### Kimai version

1.30.11 -> 2.25

### How do you run Kimai?

Virtual Server or alike

### Which PHP version are you using?

8.2

### Logfile

  An exception occurred while executing a query: SQLSTATE[42S22]: Column not found: 1054 Unknown column 'option' in 'where clause'

**Comment by kevinpapst** [comment]
Read and apply https://github.com/kimai/kimai/blob/main/UPGRADING.md
afterward try again

**Reply by mr2017x** [reply]
Hi,

Thanks for your fast reply.

So I need to upgrade kimai1.30 to 2. In place? But for what is then the migration Skript?

I thought I can migrate from install a with old to install b with new kimai ans import then data with that module?

**Reply by kevinpapst** [reply]
![image](https://github.com/user-attachments/assets/b727a4bf-d0d2-4974-aad2-67e99dd33d74)

**Comment by mostafaqanbaryan** [comment]
I know this is old, but the problem is still there.
There is no `option` column in `configuration` table.
So maybe we should first upgrade to a lower version?

**Reply by kevinpapst** [reply]
This has nothing todo with the configuration table. The error happens, because Doctrine needs the correct database version. If you do not provide one, it will try to detect it and that can can fail with this error message.

You can upgrade from any version, you just need a correct DATABASE_URL including the `serverVersion`, see answers above

**Reply by mostafaqanbaryan** [reply]
Thank you. I'm using `mysql` and have set the `serverVersion` like this:

```
- DATABASE_URL=mysql://user:pass@database/kimai?charset=utf8mb4&serverVersion=8.3.0
```

And the version came from: "SELECT VERSION();".

I'm using `ImportBundle-2.19.0` like this:

```
bin/console kimai:import:v1 --global --timezone="Asia/Tehran" --language="fa" --country="IR" --currency="IRR" --prefix="kimai2_" "mysql://user:pass@old_database:3306/kimai?charset=utf8mb4&serverVersion=5.7.34" "random_pass"
```

Still getting the same error:
```
SQLSTATE[42S22]: Column not found: 1054 Unknown column 'option' in 'where clause'
```

Am I doing something wrong?

**Reply by kevinpapst** [reply, marked as answer]
You are talking about the migration, which is the process of running `bin/console kimai:update`.
Now suddenly you use `kimai:import:v1`.
This is an entirely different topic.

I bet you don't need that. Which Kimai version are you using?
https://www.kimai.org/documentation/versions.html

The import script is for installations which are older than 6 years.

**Reply by mostafaqanbaryan** [reply]
> You are talking about the migration, which is the process of running `bin/console kimai:update`. Now suddenly you use `kimai:import:v1`. This is an entirely different topic.
> 
> I bet you don't need that. Which Kimai version are you using? https://www.kimai.org/documentation/versions.html
> 
> The import script is for installations which are older than 6 years.

Oh, I that was my bad. I'm using 1.30.11 stable (Ayumi) and I think `kimai:update` should work on that.

**Reply by kevinpapst** [reply]
Thanks for confirming 👍 

That was probably the initial issue that @mr2017x posted as well. It never occured to me that the wording would refer to the import script. That's why it is so important to share as many infos as possible with such issues.

**Reply by mostafaqanbaryan** [reply]
> Thanks for confirming 👍
> 
> That was probably the initial issue that @mr2017x posted as well. It never occured to me that the wording would refer to the import script. That's why it is so important to share as many infos as possible with such issues.

Thank you for clearing that up. That was a big help!
